### ISSUE TITLE
All.features execution produces "Uncaught Error: There is already a parameter type with name"

### ISSUE THREAD

**Issue by pravynandas** [badeball/cypress-cucumber-preprocessor#298]
Cypress: 3.7.0
cypress-cucumber-preprocessor: 1.17.0
Cypress mode: UI/Electron

defineParameterType() is working as expected when one single feature file (CustomParameterTypes.feature) is executed (which uses that defineParameterType). 

![image](https://user-images.githubusercontent.com/4099637/71736441-0589c400-2e1f-11ea-99a8-cedd928bf113.png)

However when the same feature is executed with All.features file below error produced

![image](https://user-images.githubusercontent.com/4099637/71736478-22be9280-2e1f-11ea-9532-7e1917503d91.png)

Note: Below test used the definedParameterType (ordinal/note) example given in Readme.

**Comment by pravynandas** [comment]
Today I have moved *.feature and All.features from cypress/integration/Features/ folder to cypress/integration/ and getting following error when running "All.features"

TypeError: Cannot read property 'name' of undefined
    at C:\Users\rnandap\Repo\FundMasterApp\ui\node_modules\cypress-cucumber-preprocessor\lib\featuresLoader.js:99:66
    at Array.map (<anonymous>:null:null)
    at module.exports (C:\Users\rnandap\Repo\FundMasterApp\ui\node_modules\cypress-cucumber-preprocessor\lib\featuresLoader.js:97:6)
    at Stream.end (C:\Users\rnandap\Repo\FundMasterApp\ui\node_modules\cypress-cucumber-preprocessor\lib\index.js:21:18)
    at _end (C:\Users\rnandap\Repo\FundMasterApp\ui\node_modules\through\index.js:65:9)
    at Stream.stream.end (C:\Users\rnandap\Repo\FundMasterApp\ui\node_modules\through\index.js:74:5)
    at DestroyableTransform.onend (C:\Users\rnandap\Repo\FundMasterApp\ui\node_modules\readable-stream\lib\_stream_readable.js:577:10)
    at Object.onceWrapper (events.js:282:20)
    at DestroyableTransform.emit (events.js:199:15)
    at endReadableNT (C:\Users\rnandap\Repo\FundMasterApp\ui\node_modules\readable-stream\lib\_stream_readable.js:1010:12)
    at processTicksAndRejections (internal/process/task_queues.js:81:17)

**Comment by pravynandas** [comment]
I realize one empty feature file causing 'All.features' or 'Run all specs' (on Cypress UI) to fail with above error. Is this expected behavior?

**Comment by alastair-todd** [comment]
@pravynandas my understanding is that the .features feature is meant to be used with `cypress run` only.

The cypress gui is for focus-based development testing

If you want to run all tests why use the gui? Just use `run` its faster.

**Comment by lgandecki** [comment]
@alastair-todd is correct. 

But can you, @pravynandas , please confirm that the issue is with:

All.features using gui

but not with
All.features using run?
and not with
running all features using the GUI but not having the .features file?

**Comment by EkkoLin** [comment]
is there any update for this story? I'm encountering a similar issue!

**Comment by FrancisBourgault** [comment]
This happen even when using "cypress run". I tested with "--headless" , and it still happens.

In my case, the problem is that I have two test that import the same common file, in which my custom parameter type is defined.

I believe it may be possible to avoid the issue by replicating "#pragma once" from c++. Basically, put all of your parameter type definition in a if, then use a boolean to enter that if only once.

I haven't tested it, but the boolean should only reset when the test runner is closed.

This workaround doesn't change the fact that some import (that cannot be protected by the if) may still break the test execution.

**Comment by maga-polito** [comment]
any update on this, I'm having same problems on cypress open and run

**Comment by badeball** [comment]
Due to personal reasons, the previous maintainers of this package are stepping down and handing the reigns over to me, a long-time contributor to the project and a user of it myself. This is a responsibility I'm very excited about. Furthermore, I'd like to thank @lgandecki ++ for all the work that they've done so far.

Read more about the transfer of ownership [here](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689).

The repository has however moved and all outstanding issues are being closed. This is not a reflection of the perceived importance of your reported issue. However, if after upgrading to the new version, you still find there to be an issue, feel free to open up another ticket or comment below. Please make sure to read [CONTRIBUTING.md](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#instructions-for-logging-issues) before doing so.
